testplan/report/testing/schemas.py

```diff
@@ -81,6 +81,8 @@ class TestCaseReportSchema(ReportSchema):
     timer = TimerField(required=True)
     tags = TagField()
 
+    xfail = fields.Dict(allow_none=True)
```

Comment:
We definitely shouldn't be serializing this field - the XFAIL information will already be encoded in the status field. As per above comment, I don't think that a separate xfail attribute should exist in the report at all.